### PR TITLE
[FIX] stock_account: decrease number of queries in _run_fifo_vacuum

### PR DESCRIPTION
**Pull request by yelizariev**
_run_fifo_vacuum may take significant time. This will slow down any operations
related to stocks (Inventory Adjustment, POS Session closing, etc.) or even
block them due to timeout error. So, decreasing execution time is quite
desirable. This commit is doing it by decreasing number queries which is done by
avoiding calling `search` and using _compute_value_svl in for-loop.

### CODE REVIEW

**Comment by yelizariev** [comment]
As I remember, this drops execution time for each iteration from ~350ms to ~50 ms

File: addons/stock_account/models/product.py [hunk 1]
```diff
@@ -370,25 +370,33 @@ def _run_fifo_vacuum(self, company=None):
...
             ('remaining_qty', '<', 0),
             ('stock_move_id', '!=', False),
             ('company_id', '=', company.id),
         ], order='create_date, id')
         as_svls = []
+
+        domain = [
+            ('company_id', '=', company.id),
+            ('product_id', '=', self.id),
+            ('remaining_qty', '>', 0),
```
Comment (amoyaux) [inline addons/stock_account/models/product.py:387]: You could add a create_date >= svls_to_vacuum[0].create_date. It should never happen but we never know.

File: addons/stock_account/models/product.py [hunk 2]
```diff
@@ -440,11 +450,12 @@ def _run_fifo_vacuum(self, company=None):
...
             }
             vacuum_svl = self.env['stock.valuation.layer'].sudo().create(vals)
+            # recompute it manually to avoid using heavy _compute_value_svl
+            value_svl += corrected_value
 
             # If some negative stock were fixed, we need to recompute the standard price.
-            product = self.with_company(company.id)
-            if product.cost_method == 'average' and not float_is_zero(product.quantity_svl, precision_rounding=self.uom_id.rounding):
-                product.sudo().with_context(disable_auto_svl=True).write({'standard_price': product.value_svl / product.quantity_svl})
+            if product.cost_method == 'average' and not float_is_zero(quantity_svl, precision_rounding=self.uom_id.rounding):
```
Comment (amoyaux) [inline addons/stock_account/models/product.py:457]: You can't just put it outside the loop and recompute it only at the end instead. I don't think we need it inside the loop.

**Comment by yelizariev** [comment]
@amoyaux done

**Comment by yelizariev** [comment]
@amoyaux I think we need to merge it to v14 instead https://github.com/odoo/odoo/pull/69040#issuecomment-818501176

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**